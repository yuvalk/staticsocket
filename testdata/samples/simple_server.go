package main

import (
	"net"
	"net/http"
)

const serverPort = ":8080"

func main() {
	// HTTP server on port 3000
	http.ListenAndServe(":3000", nil)

	// TCP listener with constant
	listener, _ := net.Listen("tcp", serverPort)
	defer listener.Close()

	// HTTPS server
	http.ListenAndServeTLS(":8443", "cert.pem", "key.pem", nil)

	// UDP listener
	conn, _ := net.ListenUDP("udp", &net.UDPAddr{Port: 9090})
	defer conn.Close()
}