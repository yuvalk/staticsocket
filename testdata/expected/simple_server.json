{
  "sockets": [
    {
      "type": "ingress",
      "protocol": "http",
      "process_name": "samples",
      "source_file": "testdata/samples/simple_server.go",
      "source_line": 11,
      "function_name": "unknown",
      "listen_port": 3000,
      "listen_interface": "0.0.0.0",
      "is_resolved": true,
      "raw_value": ":3000",
      "pattern_match": "http.ListenAndServe"
    },
    {
      "type": "ingress",
      "protocol": "tcp",
      "process_name": "samples",
      "source_file": "testdata/samples/simple_server.go",
      "source_line": 14,
      "function_name": "unknown",
      "listen_port": 8080,
      "listen_interface": "0.0.0.0",
      "is_resolved": true,
      "raw_value": ":8080",
      "pattern_match": "net.Listen"
    },
    {
      "type": "ingress",
      "protocol": "https",
      "process_name": "samples",
      "source_file": "testdata/samples/simple_server.go",
      "source_line": 18,
      "function_name": "unknown",
      "listen_port": 8443,
      "listen_interface": "0.0.0.0",
      "is_resolved": true,
      "raw_value": ":8443",
      "pattern_match": "http.ListenAndServeTLS"
    }
  ],
  "total_count": 3,
  "ingress_count": 3,
  "egress_count": 0,
  "process_name": ""
}